blocks/basic/include/gnuradio-4.0/basic/DataSink.hpp

```diff
@@ -38,6 +36,12 @@ concept StreamCallback = std::invocable<T, std::span<const V>> || std::invocable
 template<typename T, typename V>
 concept DataSetMatcher = std::invocable<T, gr::DataSet<V>> && std::is_same_v<std::invoke_result_t<T, gr::DataSet<V>>, bool>;
 
+template<typename T>
+concept DataSinkOrDataSetSinkLike = gr::meta::is_instantiation_of<T, DataSink> || gr::meta::is_instantiation_of<T, DataSetSink>;
```

Comment:
This is not really a 'concept' in terms of 'duck-typing' but basically forcing that only one of two specific implementation/instantiations are allowed. A `requires (...)` clause or template specialisation would have IMO been better since the templates below cannot be fulfilled by any other than the two given implementations.

Reply:
OK for now, but 'DataSinkOrDataSetSinkLike' should be removed in one of the upcoming PRs.